### PR TITLE
Fix use of deprecated APIs in Qt 5.15

### PR DESCRIPTION
**Pull request by rnhmjoj**
This is necessary to build Antimony with Qt 5.15, older versions of Qt are being deprecated and removed by distributions.

### CODE REVIEW

File: app/canvas/info.h [hunk 1]
```diff
@@ -8,8 +8,8 @@ class Datum;
 
 struct CanvasInfo
 {
-    QMap<Node*, QPointF> inspector;
-    QMap<Datum*, QPointF> subdatum;
+    QMultiMap<Node*, QPointF> inspector;
+    QMultiMap<Datum*, QPointF> subdatum;
```
Comment (mkeeter) [inline app/canvas/info.h]: I don't understand these changes – [`QMap`](https://doc.qt.io/qt-6/qmap.html) is still supported, as is [`operator[]`](https://doc.qt.io/qt-6/qmap.html#operator-5b-5d).

**Comment by mkeeter** [comment]
This looks reasonable, except the usage of `QMultiMap` instead of `QMap` - can you explain what that's about / what errors you were seeing before those changes?  It changes the semantics of the operations, so it's not a drop-in replacement.

File: app/canvas/info.h (hunk 1 above)
Comment (rnhmjoj) [reply app/canvas/info.h]: QMap is still supported but the [unite](https://doc.qt.io/qt-5/qmap-obsolete.html#unite) method (used [here](https://github.com/mkeeter/antimony/blob/ee525bbdad34ae94879fd055821f92bcef74e83f/app/canvas/info.cpp#L5)) has been deprecated and moved to this QMultiMap subclass, that handle multiple values per-key. The `[]` operator has been made private as they want you to use `value()` or `replace()` instead.

**Comment by rnhmjoj** [comment]
> It changes the semantics of the operations, so it's not a drop-in replacement.

I tried following the Qt guide regarding the deprecation and migrating the code from `QMap` to `QMultiMap` assuming multiple values per-key weren't allowed before. If this is not true, I don't how to port the code, because I'm not sure about what `map[key]` would do before, with multiple values.

**Comment by mkeeter** [comment]
I don't believe I'm using multiple values per key here.  The [list of obsolete members](https://doc.qt.io/qt-5/qmap-obsolete.html) doesn't include `operator[]`.

**Comment by rnhmjoj** [comment]
> I don't believe I'm using multiple values per key here.

Ok, then it should be fine. By the way, I build Antimony with Qt 5.15 and tested it by opening some .sb files and playing around a bit: everything seems fine.

> The [list of obsolete members](https://doc.qt.io/qt-5/qmap-obsolete.html) doesn't include operator[].

As I said in the comment above, I had to change `QMap` to `QMultiMap` because `unite()` has been moved into `QMultiMap` and deprecated in `QMap`.

**Comment by mkeeter** [comment]
Aha, I missed your comment about `unite`.

I believe that it's only used to join two maps – and not for the multi-value behavior – so could be replaced [here](https://github.com/mkeeter/antimony/blob/ee525bbdad34ae94879fd055821f92bcef74e83f/app/canvas/info.cpp#L3-L7) with
```cpp
void CanvasInfo::unite(const CanvasInfo& other)
{
    for (const auto i: other.inspector) {
        inspector.insert(i);
    }
    for (const auto s: other.subdatum) {
        subdatum.insert(s);
    }
}
```
(untested)

It looks like `operator[]` is still public for `QMap`, so if you modify the `CanvasInfo::unite` implementation (and switch back to `QMap`), you could go back to using it.

**Comment by rnhmjoj** [comment]
I've found a simpler solution: I missed that 5.15 overloaded the `insert()` method to take another `QMap`: assuming that no duplicate keys exists it works exactly like `unite()`.

**Comment by mkeeter** [comment]
Thanks!

**Comment by rnhmjoj** [comment]
Thank *you*. I really appreciate the quick response even if the project is in low mantainenance mode.

**Pull request merged**